### ISSUE TITLE
z/OS 2.3 "IEE479W MASTER SCHEDULER ABEND"

### ISSUE THREAD

**Issue by rpinion** [SDL-Hercules-390/hyperion#93]
**`IEF403I MSTJCLXX - STARTED`**
**`IEE479W MASTER SCHEDULER ABEND 0A0, DUMPED, REIPL - CODE 1Z`** 

I have a legal copy of ADCD z/OS 2.3. I am a zPDT customer, and wanted to try z/OS 2.3 on your version of Hercules.  The out of the box ADCD IPL's fine on zPDT.  However, I have customized my z/OS 2.3 DASD images to better reflect my current, z/OS 2.2, configuration.  Since my zPDT instance is in use by other users, I'm not at liberty to change, IPL, and test at will.  Hence, my desire to use Hercules.  I saw your post a few weeks ago about your fork of Hercules supporting z/OS 2.3, so I decided to give it a try.

Under your fork of Hercules, the Master Scheduler abends with S0A0 code 1Z.  I have checked SYS1.PARMLIB(MSTJCL00) to insure all of the data sets are cataloged in the master catalog, which they are.  In addition, I hard coded volser and unit=3390 on the data sets, with the same results.

Yesterday, while Google'ing, I ran across an entry in Git, sometime in 2017, regarding Hercules not properly filling in CU/UCB information.  Which might be related to the message IEF462I for each of the DD names in MSTJCL00 .  I checked and double checked that the DASD addresses are in the IODF config, the unit type is correct, and that they are online at IPL.

FWIW, z/OS 2.2 IPL's and runs fine on your fork of Hercules.

**Comment by Fish-Git** [comment]
You are going to hate me for saying this Richard, but this does not sound like a Hercules problem.  This sounds like a z/OS configuration/customization issue.

I have on good authority that z/OS 2.3 -- even heavily customized ones -- IPL and run just fine with my latest SDL Hyperion 4.0 release ([build 9176](http://www.softdevlabs.com/hyperion.html#9176)).

Perhaps your configuration file isn't defined properly? It's important to read the Release Notes with each new release of Hercules for changes that were made that might impact its usage.

It might also help some if you would post the contents of your Hercules configuration file and the complete contents (_from beginning to end_) of your Hercules log too. Submitting a problem report with incomplete information isn't very helpful and wastes valuable time.

I am closing this issue as "Invalid" ("Won't fix").  Please feel free to re-open this issue again if you feel it should be (but only after providing the mentioned missing configuration and log files!)

Thanks.

**Comment by rpinion** [comment]
If someone else is doing it, then it must be something I've done.  I'll continue to work on it.  Thanks.Richard and Vickie Pinion--- notifications@github.com wrote:From: Fish-Git <notifications@github.com>To: Fish-Git/hyperion <hyperion@noreply.github.com>Cc: rpinion <rpinion@netscape.com>,  Author <author@noreply.github.com>Subject: Re: [Fish-Git/hyperion] z/OS 2.3 "IEE479W MASTER SCHEDULER ABEND" (#93)Date: Tue, 01 May 2018 03:10:42 -0700You are going to hate me for saying this Richard, but this does not sound like a Hercules problem.  This sounds like a z/OS configuration/customization issue.
I have on good authority that z/OS 2.3 -- even heavily customized ones -- IPL and run just fine with my latest SDL Hyperion 4.0 release (build 9176).
Perhaps your configuration file isn't defined properly? It's important to read the Release Notes with each new release of Hercules for changes that were made that might impact its usage.
It might also help some if you would post the contents of your Hercules configuration file and the complete contents (from beginning to end) of your Hercules log too. Submitting a problem report with incomplete information isn't very helpful and wastes valuable time.
I am closing this issue as "Invalid" ("Won't fix").  Please feel free to re-open this issue again if you feel it should be (but only after providing the mentioned missing configuration and log files!)
Thanks.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




 Netscape.  Just the Net You Need.

**Comment by Fish-Git** [comment]
I was notified that someone else is also experiencing the same problem, so I am re-opening this issue.

I am not going to classify it (label it) as a "bug" yet (since the underlying causes is still unknown), but have added a new "Researching..." label since the problem, whatever it is, is being actively looked into by outside parties.

I will update this GitHub Issue as more information comes in.

**Comment by rpinion** [comment]
On vacation until May 28, thanks.On May 5, 2018 8:56 PM, Fish-Git <notifications@github.com> wrote:Reopened #93.

**Comment by rpinion** [comment]
Thanks Mr. Fish!On May 5, 2018 8:56 PM, Fish-Git <notifications@github.com> wrote:I was notified that someone else is also experiencing the same problem, so I am re-opening this issue.
I am not going to classify it (label it) as a "bug" yet (since the underlying causes is still unknown), but have added a new "Researching..." label since the problem, whatever it is, is being actively looked into by outside parties.
I will update this GitHub Issue as more information comes in.

**Comment by Fish-Git** [comment]
I came across [the following post](https://groups.yahoo.com/neo/groups/z1090/conversations/messages/3618) on the [z1090](https://groups.yahoo.com/neo/groups/z1090/info) Yahoo group (the official support group for IBM's zPDT product) and am wondering if maybe the lines I've marked with ">>" in the below z/PDT log extract might be a potential clue as to maybe why the ADCD version of z/OS 2.3 doesn't currently run under Hercules:

```
   CMD : 050618 12:03:48: ipl a80 parm 0a82db
   LOG : 050618 12:03:51: CPU: CPU reset
>> LOG : 050618 12:03:51: CPU: Accepted code 7
   LOG : 050618 12:03:52: CPU: CPU reset
>> LOG : 050618 12:03:52: CPU: Accepted code 7
>> INFO: 050618 12:03:52: MIP: AWSMIP008I zArchitecture IPL mode (ZARCH_ONLY=ON)
   LOG : 050618 12:03:53: CPU: IPL started on CPU 0
   LOG : 050618 12:03:53: CPU: IPR started for CPU 0
   LOG : 050618 12:03:53: CPU: CPU reset
   LOG : 050618 12:03:53: CPU: IPR complete for CPU 0
   LOG : 050618 12:03:54: CPU: IPL bootstrap successful for device A80
>> LOG : 050618 12:03:54: CPU: Accepted code 0
   LOG : 050618 12:03:54: CPU: Running CPU
>> LOG : 050618 12:03:54: CPU: Short PSW Change 0008000080000652
```

I'm wondering if the _**CPU: Accepted code 7**_ and _**CPU: Accepted code 0**_ lines might be the z/PDT logging its acceptance of an undocumented SIGP (Signal Processor) order. The latest Principles of Operation manual doesn't document what SIGP orders 0 and 7 are. They're both labeled "unassigned".

The line _**MIP: AWSMIP008I zArchitecture IPL mode (ZARCH_ONLY=ON)**_ and _**CPU: Short PSW Change 0008000080000652**_ lines are even _more_ intriguing IMO. The latest [**IBM zPDT Reference manual (SG24-8205-03)**](http://www.redbooks.ibm.com/redbooks/pdfs/sg248205.pdf) documents its `cpuopt` statement on page 252 as follows:

> **13.3 cpuopt statement**
>
> The ZARCH_ONLY parameter is new with zPDT GA8, defaults to YES, and is the standard mode of operation for IBM z14 systems. It causes the z14 to IPL in normal zArchitecture mode. Specifying ZARCH_ONLY=NO causes zPDT to IPL in ESA390 mode, but otherwise operate as a z14 system. This is a non-standard configuration that might be useful with older operating systems that are not designed to IPL in zArchitecture mode. Use of this option creates an environment that is not supported by IBM, and should be used with caution. (The ZARCH_ONLY function is sometimes documented as the CZAM facility.) IPLing an older z/OS operating system (that does not support IPL in zArchitecture mode) on a z14 can produce disabled wait state 19. The ZARCH_ONLY=NO option might be useful in such situations. The default is ZARCH_ONLY=YES.

Perhaps the reason why z/OS 2.3 runs okay on the z/PDT but doesn't on Hercules is because the z/PDT is emulating a z14 which supports both the "CZAM" (Configuration-z/Architecture-Architectural-Mode,
 STFLE bit 138) and "ESA/390-Compatability-Mode" (STFLE bit 168) facilities, which Hercules doesn't support yet.

Maybe the only thing we need to do is code the missing support for the CZAM and 390 Compatibility facilities?

After all, if I'm not mistaken, the ADCD is designed specifically for running on the z/PDT. I would not be surprised at all if it was purposely built (configured) to expect/require that both facilities be present, whereas those who are currently successfully running z/OS 2.3 under Hercules built it themselves (i.e. those successfully running z/OS 2.3 under Hercules are _not_ using the ADCD but rather a copy of z/OS 2.3 they obtained elsewhere).

Anyway, it's just a thought. &nbsp; `:)`

When I find the time I'll try to see how difficult if would be to code support for the CZAM and 390 Compatibility Mode facilities. (On the face of it, it doesn't sound too tough, but in my experience the devil is always in the details.)

**Comment by rpinion** [comment]
OK, thank you.  On holiday, so unable to work on Hercules until May 28.On May 7, 2018 7:01 PM, Fish-Git <notifications@github.com> wrote:I came across the following post on the z1090 Yahoo group (the official support group for IBM's zPDT product) and am wondering if maybe the lines I've marked with ">>" in the below z/PDT log extract might be a potential clue as to maybe why the ADCD version of z/OS 2.3 doesn't currently run under Hercules:
   CMD : 050618 12:03:48: ipl a80 parm 0a82db
   LOG : 050618 12:03:51: CPU: CPU reset
>> LOG : 050618 12:03:51: CPU: Accepted code 7
   LOG : 050618 12:03:52: CPU: CPU reset
>> LOG : 050618 12:03:52: CPU: Accepted code 7
>> INFO: 050618 12:03:52: MIP: AWSMIP008I zArchitecture IPL mode (ZARCH_ONLY=ON)
   LOG : 050618 12:03:53: CPU: IPL started on CPU 0
   LOG : 050618 12:03:53: CPU: IPR started for CPU 0
   LOG : 050618 12:03:53: CPU: CPU reset
   LOG : 050618 12:03:53: CPU: IPR complete for CPU 0
   LOG : 050618 12:03:54: CPU: IPL bootstrap successful for device A80
>> LOG : 050618 12:03:54: CPU: Accepted code 0
   LOG : 050618 12:03:54: CPU: Running CPU
>> LOG : 050618 12:03:54: CPU: Short PSW Change 0008000080000652

I'm wondering if the CPU: Accepted code 7 and CPU: Accepted code 0 lines might be the z/PDT logging its acceptance of an undocumented SIGP (Signal Processor) order. The latest Principles of Operation manual doesn't document what SIGP orders 0 and 7 are. They're both labeled "unassigned".
The line MIP: AWSMIP008I zArchitecture IPL mode (ZARCH_ONLY=ON) and CPU: Short PSW Change 0008000080000652 lines are even more intriguing IMO. The latest IBM zPDT Reference manual (SG24-8205-03) documents its cpuopt statement on page 252 as follows:

13.3 cpuopt statement
The ZARCH_ONLY parameter is new with zPDT GA8, defaults to YES, and is the standard mode of operation for IBM z14 systems. It causes the z14 to IPL in normal zArchitecture mode. Specifying ZARCH_ONLY=NO causes zPDT to IPL in ESA390 mode, but otherwise operate as a z14 system. This is a non-standard configuration that might be useful with older operating systems that are not designed to IPL in zArchitecture mode. Use of this option creates an environment that is not supported by IBM, and should be used with caution. (The ZARCH_ONLY function is sometimes documented as the CZAM facility.) IPLing an older z/OS operating system (that does not support IPL in zArchitecture mode) on a z14 can produce disabled wait state 19. The ZARCH_ONLY=NO option might be useful in such situations. The default is ZARCH_ONLY=YES.

Perhaps the reason why z/OS 2.3 runs okay on the z/PDT but doesn't on Hercules is because the z/PDT is emulating a z14 which supports both the "CZAM" (Configuration-z/Architecture-Architectural-Mode,
STFLE bit 138) and "ESA/390-Compatability-Mode" (STFLE bit 168) facilities, which Hercules doesn't support yet.
Maybe the only thing we need to do is code the missing support for the CZAM and 390 Compatibility facilities?
After all, if I'm not mistaken, the ADCD is designed specifically for running on the z/PDT. I would not be surprised at all if it was purposely built (configured) to expect/require that both facilities be present, whereas those who are currently successfully running z/OS 2.3 under Hercules built it themselves (i.e. those successfully running z/OS 2.3 under Hercules are not using the ADCD but rather a copy of z/OS 2.3 they obtained elsewhere).
Anyway, it's just a thought.   :)
When I find the time I'll try to see how difficult if would be to code support for the CZAM and 390 Compatibility Mode facilities. (On the face of it, it doesn't sound too tough, but in my experience the devil is always in the details.)

**Comment by Fish-Git** [comment]
Richard,

FYI: if you are going to reply by email (as opposed to replying directly via the[ GitHub Issues user interface](https://github.com/Fish-Git/hyperion/issues/93), which is the _preferred_ way to reply), it would be appreciated if you would please _delete/remove_ all previous text so I don't have to manually edit your replies. It's _not_ a big deal, so you can continue to reply as you have been if you want, and I will continue to manually edit your replies as needed, but it would just be nice if I didn't have to do that.

Thanks.

Enjoy your vacation! &nbsp; We'll see you when you get back. &nbsp;`:)`

**Comment by Juergen-Git** [comment]
Hi Fish

just a quick heads up to avoid unnecessary efforts: I heard from a knowledgeable source that CZAM and 390 Compat mode are not part of this problem, and also, that those SIGP instructions using unassigned command codes 0 and 7 most probably are not involved. Furthermore, that source confirms that all three current Hercules versions (SDL, Hyperion and Spinhawk) can run the z/OS 2.3 ADCD, after applying a small change.

Cheers
Jürgen

**Comment by rpinion** [comment]
What is the change?--- notifications@github.com wrote:From: Jürgen Winkelmann <notifications@github.com>To: Fish-Git/hyperion <hyperion@noreply.github.com>Cc: rpinion <rpinion@netscape.com>,  Author <author@noreply.github.com>Subject: Re: [Fish-Git/hyperion] z/OS 2.3 "IEE479W MASTER SCHEDULER ABEND" (#93)Date: Fri, 11 May 2018 12:29:05 -0700Hi Fish
just a quick heads up to avoid unnecessary efforts: I heard from a knowledgeable source that CZAM and 390 Compat mode are not part of this problem, and also, that those SIGP instructions using unassigned command codes 0 and 7 most probably are not involved. Furthermore, that source confirms that all three current Hercules versions (SDL, Hyperion and Spinhawk) can run the z/OS 2.3 ADCD, after applying a small change.
Cheers
Jürgen

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.


  
  





 Netscape.  Just the Net You Need.

**Comment by Fish-Git** [comment]
> What is the change?

I was going to ask the same thing, Richard!

**Comment by mainframewiz01** [comment]
Hi Fish,

I am a licenced zPDT 2.2 user and have been trying(out of curiosity) to use the ADCD volumes with Hercules Hyperion. I've been able to IPL the system but unable to get past the abend 0A0 (IEE479W MASTER SCHEDULER ABEND). From what I've read and concluded (based on your comments) is that the ADCD version of the volumes might not be compatible with Hercules, correct?

People on different forums have posted different solutions to identify/highlight this issue but none of them seems to work(at least not with the version of the volumes that I've got)! Has there been any progress on this? 

Also, is there a way to legally acquire these volumes(the ones that work with Hyperion)?

Thanks :)

**Comment by sford1950** [comment]
David.

We are a licensed zPDT shop and have z/OS 2.2 and z/OS 2.3 working fine as well as zVM 6.4 
We also want to use Hercules and tried it with SD-Hercules on OpenSuse Leap 42.3 x64 and I am seeing the same abend Richard is seeing on z/OS 2.3 .  We need SSH working on z/OS 2.2 or z/Os 2.3 on Hercules. But the abend - MSTJCL00 - ABEND0A0 is stopping us. Can we help out in any way ?

Scott

**Comment by Juergen-Git** [comment]
Well, Scott, I still don't have the slightest idea as to what the solution might be -- if one exists at all. Maybe, everything is hearsay only?

However, you are mentioning z/VM 6.4. Just of curiousity: Did you try this one on Hercules? And if so, does it work?

Jürgen

**Comment by sford1950** [comment]
Jurgen:

Good question, i am not sure. I dont think CSF will work on SDL_Hercules. I am not sure.
But the issue still stands for z/OS 2.2 and 2.3 , I dont know if anyone has z/OS 2.2 and SDL-Hercules working, If they do is there a sample of the working Hercules cfg...

Scott

**Comment by Juergen-Git** [comment]
Well, as I said, there is no direct solution I’d be aware of. However, fwiw, let me summarize what I know about the issue. Note however, that all of this is mostly esoteric. Heard this, seen that and drawn daring conclusions from it. So, don’t take anything of the following for fact:

Symptom: IEE479W MASTER SCHEDULER ABEND 0A0, DUMPED, REIPL - CODE 1Z

This message basically says that the allocation of a dataset needed by MSTRJCL failed. This can mean everything, for example the data set doesn’t exist, or it isn’t cataloged, or it is cataloged but the catalog entry doesn’t point to the correct DASD, or the DD statement names incorrect UNIT and/or VOL=SER, and so on.

Many have tried weirdest changes to MSTRJCL but apparently none of them succeeded. Taking a standalone dump immediately reveals the reason: The eligible device table (EDT) of the system doesn’t contain a single DASD and thus it’s clearly impossible to allocate any DASD data set at all. So, the real question is: Why were the DASDs of the active Hercules configuration not used to populate the EDT? 

Potential culprits to look at in the first place might be during IPL IEAIPL08, IEAIPL03 and IEAIPL18, or during NIP IEAVNPB2 and IEAVNP88. However, looking at those doesn’t give an immediate clue.

On the other hand, instead of going from IPL forward, one can also go backward from the point where the initiator to run IEEMB860 gets attached, trying to find the location that does the last change to the EDT. And bingo! The location is easy to find and staring a bit at it one finds a branch that’s only taken in case of DASD devices. This branch ends up not adding Hercules DASDs to the EDT. Consequently, not following this branch should add the DASDs to the EDT and thus allow MSTRJCL initialization to complete normally. I heard sayings that this branch can be found at location X’0406’ in IEFIBER3 and thus ZAPping this location to read X’ A7040070’ would bring the system into smooth sailing. But as I said: Don’t try this at home ;-)

**Comment by sford1950** [comment]
Juergen,

Thank you i will give the zap a whirl ..we are a development shop so no problem in doing that ..
I will let you know what happens..

Scott

**Comment by jiflemon** [comment]
Juergen,

Can I just check that offset, and ask what branch is there before zapping....? IEFIBER3 X'0460' has X'5830F000' on my system.

**Comment by Juergen-Git** [comment]
IEFIBER3 X'0460'? No idea what is or should be there...

**Comment by Juergen-Git** [comment]
I'd appreciate if anyone who succeeded or failed using the above guesswork would provide feedback. That simply because I'd like to remove it, if it doesn't work, so to not lead people on yet another invalid path (as there are enough ones around already).

**Comment by jiflemon** [comment]
Zap works. z/OS 2.3 IPL without the Abend 0A0

**Comment by Juergen-Git** [comment]
Thanks! So, I consider the original request answered in some sense. Of course the root cause of the problem still remains unknown, but that's probably for another day.

**Comment by Fish-Git** [comment]
I am going to close this issue now because I see no reason to keep it open any longer.

The problem/bug in Hercules that is causing z/OS 2.3 to not add Hercules's dasds to its EDT is unknown and will very likely remain unknown.  We don't know how IBM coded their z/OS, and thus we don't know what it's complaining about (i.e. we don't know, and will likely never(?) know, what it doesn't like about our dasd).

But since a viable "workaround" seems to have developed (which several people have confirmed works okay), I don't believe there's anything more we can do to resolve this issue. The documented workaround (patch) is the best we can do for now.

If/when someone more experienced and knowledgeable with z/OS (internals?) ever eventually determines exactly what z/OS is complaining about regarding our dasd, _THEN_ we can re-open this issue and fix it.

But at this moment I don't believe there's anything more we can do to resolve this issue, so I think at this time it should be closed.

**Comment by cfdonatucci** [comment]
Hi Guys, fix worked on 2.3 Is it the same fix for 2.4? 
Regards
Carlos

**Comment by cfdonatucci** [comment]
Hi guys, sorry to insist on this question. Is the same offset for 2.4? has anybody tried 2.4?
Regards
Carlos

**Comment by andysilva86** [comment]
Sorry for writing here.

Could you let me know if there is any documentation on how to do this ZAP in Hercules 4.7? How to do this bypass in Hercules, the same error is occurring in the z/OS 3.1 Test Development Image

**Comment by andysilva86** [comment]
> Use 4.8-DEV.

Hi @Fish-Git 

It worked! The 4.8-DEV worked!

However, I needed to use 'FAC ENA 054', 'FAC ENA 129', 'FAC ENA 130', 'FAC ENA 134', and 'FAC ENA 135' before IPL. Will this be implemented with the sharedvfp branch merge? I am asking because I tried the sharedvfp branch and encountered the master scheduler abend A0A, just out of curiosity.

Really thanks!!!

**Comment by Fish-Git** [comment]
> However, I needed to use 'FAC ENA 054', 'FAC ENA 129', 'FAC ENA 130', 'FAC ENA 134', and 'FAC ENA 135' before IPL.

Correct. Those are the facilities that z/OS 3.1 requires. Without any one of them it will fail to IPL. It needs them all.

> Will this be implemented with the sharedvfp branch merge?

Only those facilities which are actually implemented will ever be enabled. Duh.

> I am asking because I tried the `sharedvfp` branch just out of curiosity and encountered the master scheduler abend A0A.

Expected. The `sharedvfp` branch only contains vector instruction development.

**Comment by ARandham** [comment]
> Use 4.8-DEV.

Hi Fish,
Which commit(s) in 4.8-DEV resolved the issue?

Andrew

**Comment by ARandham** [comment]
Never mind, looks like it was the almost "cometic" commit 9bf7a4f that corrected the issue.

Andrew

**Comment by prenard** [comment]
Hello All.
I would like to run z/OS 3.1 ADCD on Hercule.
I understant that I need to use 4.8-DEV - Where can I download a pre-built windows version of 4.8-DEV ?
Thanks for yourn help

**Comment by Fish-Git** [comment]
> I understand that I need to use 4.8-DEV - Where can I download a pre-built windows version of 4.8-DEV ?

Pre-built versions of Hercules for Windows are only available for official releases. There are no pre-built versions of Hercules for Windows _(nor for Linux or any other host operating system for that matter!)_ for development versions of Hercules. For development versions of Hercules you have to build Hercules for yourself.

The fastest and easiest way to do this is to use Bill Lewis's _(one of our Hercules developers)_ "Hercules Helper" script. It completely automates building Hercules.

He has two versions available: one for non-Windows and a separate one just for Windows:

* **_non_-Windows** Hercules Helper:
  https://github.com/wrljet/hercules-helper

* _**Windows**_ Hercules Helper:
  https://github.com/wrljet/hercules-helper-windows

Simply clone his repository and then enter the appropriate command along with the needed options, and in a few short minutes you will have your very own Hercules! His README for each explains how to use it:

* Instructions for **_non_-Windows** Hercules Helper:
  https://github.com/wrljet/hercules-helper/blob/master/README.md

* Instructions for _**Windows**_ Hercules Helper:
  https://github.com/wrljet/hercules-helper-windows/blob/master/README.md

**Comment by prenard** [comment]
Thanks.
I was able to build 4.8-Dev using "Hercules Helper".
I confirm that it is now possible to IPL z/OS 3.1 on 4.8-DEV.

**Comment by Fish-Git** [comment]
> But I tried to IPL z/OS 3.1 as a z/VM 7.3 guest and I got Wait State 7B / 22 - See new issue #694

Yes, that will happen when z/OS is IPLed as a guest under z/VM and z/VM fails to pass all of its z/Arch facility bit settings on to the guest. Why it does this I don't know. Nevertheless, there is a workaround.

One of our developers has a z/VM EXEC that patches z/VM on the fly during a z/OS guest IPL to force all of the z/VM host's facility bits to be passed on to the guest so that the z/OS guest properly sees _ALL_ of the available host facilities, thereby allowing e.g. z/OS 3.1 to successfully IPL. (He also has some that allows guests to use ZIIP processors too.)

* [zOS-zVM.zip](https://github.com/user-attachments/files/18000550/zOS-zVM.zip)

Place the 3 EXECs somewhere where your guest(s) can get to them _(I personally place them on my system's 'C' disk so all guests can see them)_, and then use the `STFLEFIX` exec to IPL your guest with instead. _(See e.g. my own `IPLZOS31` exec)_. That is to say, when you want to IPL your z/OS guest, _don't_ use the normal "CP IPL ..." command. Instead, use the appropriate "STFLEFIX" command instead. We've been using this technique for the past several versions of z/OS and it works great.

Hope that helps!

**Comment by prenard** [comment]
Thanks for your help.

Using MAINT730 user, I have copied exec files to C disk (02CC / PMT2CC).

C disk was not mounted by default on my guest, so I had to issue "ACC 02CC C" to mount it.

At this point, I have the following disks configuration on the guest:

query disk                                                                      
LABEL  VDEV M  STAT   CYL TYPE BLKSZ   FILES  BLKS USED-(%) BLKS LEFT  BLK TOTAL
PMT2CC 2CC  C   R/O    10 3390 4096       18        216-12       1584       1800
MNT190 190  S   R/O   207 3390 4096      683      15622-42      21638      37260
MNT19E 19E  Y/S R/O   500 3390 4096     1175      74615-83      15385      90000
Ready; T=0.01/0.01 09:43:32                                                     


When I issue "STFLEFIX ZIIPFIX IPL IPLDEV 0Ab7 PARMDEV 0Ab9 PARMXX WS" I get:

STFLEFIX ZIIPFIX IPL IPLDEV 0Ab7 PARMDEV 0Ab9 PARMXX WS                   
HCPQVD022E A virtual device number was not supplied or it was invalid.    
HCPQVD022E A virtual device number was not supplied or it was invalid.    
Ready(01022); T=0.02/0.02 09:43:51                                        
                                                                          
Do I have to mount (ACCESS command) 0AB7 & 0AB9 before issuing STFLEFIX exec ?

**Comment by prenard** [comment]
With PREPARE & LIST parameters:

STFLEFIX ZIIPFIX prepare ipldev 0AB7 PARMDEV 0AB9 PARMXX WS list  
PREPARE overwrites all other parameters which are thus ignored.   
About to IPL z/OS and produce a STFLE trace, followed by a        
re-IPL of CMS.  A subsequent 2nd run of STFLEFIX EXEC can         
use that trace printer file to produce a z/OS IPL with the        
STFLE instruction patched with the correct bits set.              
Ready(00004); T=0.05/0.06 10:19:39                                
Tracing active at IPL                                             
 -> 00000000000006AE   STFLE   B2B00438 >> 00000438    CC 0       
R00000438  F3E2FFFB FCFDFE64 007CC000 00000000 06                 
R00000448  20004000 04000000 00000000 00000000                                  
GRG  0 =  0000000000000003                                                      
PRT FILE 0008 SENT FROM S0W1     PRT WAS 0008 RECS 0001 CPY  001 A HOLD   NOKEEP
Trace ended                                                                     
z/VM V7.3.0    2022-06-23 15:58                                                 
DMSACP113S A(191) not attached or invalid device address                        
Ready; T=0.11/0.12 10:20:15

**Comment by prenard** [comment]
I have created 191 V-DISK

query disk                                                                      
LABEL  VDEV M  STAT   CYL TYPE BLKSZ   FILES  BLKS USED-(%) BLKS LEFT  BLK TOTAL
LOCAL  191  A   R/W    FB 9336 4096        0          6-05        119        125
PMT2CC 2CC  C   R/O    10 3390 4096       18        217-12       1583       1800
MNT190 190  S   R/O   207 3390 4096      683      15622-42      21638      37260
MNT19E 19E  Y/S R/O   500 3390 4096     1175      74615-83      15385      90000
Ready; T=0.01/0.01 11:48:02                                                     

But now, I get the following error:

STFLEFIX ZIIPFIX IPL IPLDEV 0AB7 PARMDEV 0AB9 PARMXX WS                         
   129 +++ vmdbk = X2D(vmdbk_)                                                  
DMSREX475E Error 40 running CPUTYPES EXEC, line 129: Incorrect call to routine  
Ready; T=0.08/0.12 11:57:16                                                     
Tracing active at IPL                                                           
 -> 00000000000006AE   STFLE   B2B00438 >> 00000438    CC 0                     
Store complete.                                                                 
Trace ended                                                                     

If I wait a little bit zOS IPL is starting.... But I do not know when I have to enter CP command "TERM CONMODE 3270" so the display is wrong on the console... I tried to enter it before IPLing CMS but the result is wrong !


$- *0                                                                       
     IEA371I SYS1.IPLPARM ON DEVICE 0AB9 SELECTED FOR IPL PARAMETERS        
     IEA246I LOAD   ID WS SELECTED                                          
     IEA246I NUCLST ID 00 SELECTED                                          
     IEA519I IODF DSN = SYS1.IODF99                                         
     IEA520I CONFIGURATION ID = OS390   . IODF DEVICE NUMBER = 0AB9         
     IEA091I NUCLEUS 1 SELECTED                                             
     IOS128I IPL DEVICE: 00AB7  VOLUME: A3RES1                              
     IEA370I MASTER CATALOG SELECTED IS CATALOG.Z31A.MASTER                 
     IEA247I USING IEASYSWS FOR z/OS 03.01.00 HBB77E0                       
     IEA007I STATIC SYSTEM SYMBOL VALUES                                    
      &SYSALVL.          = "2"                                              
      &SYSCLONE.         = "1A"                                             
                                                                            
                                                                            
                                                          MORE...   VM4PDT

**Comment by Fish-Git** [comment]
@prenard 

I apologize for providing incorrect information!

I do not keep the three mentioned helper EXECs (cputypes, iplziip and stflefix) on my system's C disk, I keep them on my system's S/Y disk!

<pre>
msg * Hello
<b><i> 17:33:25  * MSG FROM ZOS     : HELLO</i></b>
Ready; T=0.01/0.01 17:33:25

q disk *
LABEL  VDEV M  STAT   CYL TYPE BLKSZ   FILES  BLKS USED-(%) BLKS LEFT  BLK TOTAL
ZOS191 191  A   R/W    10 3390 4096       12         21-01       1779       1800
MNT190 190  S   R/O   207 3390 4096      691      15640-42      21620      37260
MNT19E 19E  Y/S R/O   500 3390 4096     1175      74615-83      15385      90000
Ready; T=0.01/0.01 17:33:31

l stflefix exec *
<b><i>STFLEFIX EXEC     S2</i></b>
Ready; T=0.01/0.01 17:33:59
</pre>

<pre>
msg * hello
<b><i> 17:37:55  * MSG FROM MAINT730: HELLO</i></b>
Ready; T=0.01/0.01 17:37:55

q disk *
LABEL  VDEV M  STAT   CYL TYPE BLKSZ   FILES  BLKS USED-(%) BLKS LEFT  BLK TOTAL
MNT191 191  A   R/W   200 3390 4096       44        671-02      35329      36000
MNT5E6 5E6  B   R/W    20 3390 4096      141       1594-44       2006       3600
PMT2CC 2CC  C   R/W    10 3390 4096        3        103-06       1697       1800
MNT51D 51D  D   R/W    26 3390 4096      248       1285-27       3395       4680
PMT551 551  E   R/W    40 3390 4096       13        138-02       7062       7200
MNT190 190  S   R/O   207 3390 4096      691      15640-42      21620      37260
MNT193 193  W   R/O   500 3390 4096     1126      35480-39      54520      90000
MNT19E 19E  Y/S R/O   500 3390 4096     1175      74615-83      15385      90000
PMTCF0 CF0  Z   R/W   120 3390 4096        3         25-01      21575      21600
ady; T=0.01/0.01 17:38:04

l stflefix exec *
STFLEFIX EXEC     A1
<b><i>STFLEFIX EXEC     S2</i></b>
Ready; T=0.01/0.01 17:38:19
</pre>

I sincerely apologize for the confusion. &nbsp; `:(`

**Comment by Fish-Git** [comment]
> If I wait a little bit zOS IPL is starting.... But I do not know when I have to enter CP command "TERM CONMODE 3270" so the display is wrong on the console... I tried to enter it before IPLing CMS but the result is wrong !

Take a look at my `IPLZOS31 EXEC`: do _not_ try to use your guest's CMS terminal as your z/OS master console.

Instead, always attach a separate terminal for your z/OS master console. _(Make sure your User Direct for your z/OS guest has several of them them defined, so whenever you need one for e.g. TSO, you just `DIAL ZOS'.)_

**Comment by prenard** [comment]
With zD&T, I used guest console as zOS console... So I am a little bit lost...
i am not a zVm expert. Can you detail the operational process to ipl zOS guest  with Hercules ?

**Comment by prenard** [comment]
Thanks for your help. It is better now !

Another question about z/VM guests:

 Is-it possible to run a Coupling Facility z/VMF guest under Hercules ?

I have a Parallel Syplex configutaion (2 CF guests + 2 z/OS guests) running on z/VM 7.3 under zD&T. Is-it possible to start it under Hercules ?

**Comment by Fish-Git** [comment]
> With zD&T, I used guest console as zOS console... So I am a little bit lost...

Hercules is not zD&T.

> I am not a zVM expert.

Neither am I.

> Can you detail the operational process to ipl zOS guest with Hercules?

Hercules has nothing to do with it. Hercules is a hardware emulator. It emulates a z/Architecture machine. The procedure for running (IPLing) a z/Architecture guest under z/VM on Hercules is exactly the same as running (IPLing) a z/Architecture guest under z/VM on any other z/Architecture machine (hardware), and should be well documented in IBM's manuals.

**Comment by Fish-Git** [comment]
> Is-it possible to run a Coupling Facility z/VMF guest under Hercules?

> Is-it possible to start it under Hercules?

No. Hercules does not support Coupling Facility.

If you have any further questions regarding Hercules, please post them to our primary Hercules emulator support group on Groups.IO:

* https://hercules-390.groups.io/g/group

If you need Help with running certain specific operating systems on Hercules, please ask for help on one of our other support groups dedicated to such operating systems:

* https://sdl-hercules-390.github.io/html/hercsupp.html

(be sure to scroll through the entire page; we have many support groups.)
